server/app.js

```diff
@@ -12,8 +12,8 @@ app.get('/', function (req, res) {
 
 app.get('/restaurant/:restaurant_id/:date', function(req, res) {
   db.grabTimeSlots(req.params.restaurant_id, req.params.date, function(data) {
```

Comment:
You should also be doing error handling inside this callback. There may be an error during retrieval of data from the database which you would be sending to the client regardless as data.

Reply:
For this specific function I'm not sure how to error handle it, but in my db.js there is a error handler for this imported function so I know if the error is from that function in the db.js